packages/storefront-lwc/src/modules/commerce/basket/basket.js

```diff
@@ -51,21 +57,26 @@ export default class Basket extends LightningElement {
         return this.basket.selectedShippingMethodId;
     }
 
-    connectedCallback() {
-        ShoppingBasket.getCurrentBasket()
-            .then(basket => {
-                this.basket = basket;
-                this.products = basket.products ? basket.products : [];
-                this.loading = false;
-            })
-            .catch(error => {
-                console.log('error received ', error);
-            });
+    @wire(useMutation, {
+        mutation: REMOVE_ITEM_FROM_BASKET,
+        lazy: true,
+    })
+    removeItemFromBasket;
+
+    removeHandler(event) {
+        const itemId = event.detail.itemId;
+        const variables = { itemId };
+        this.removeItemFromBasket.mutate({ variables }).then(() => {
+            this.basket = {
+                ...this.basket,
+                ...this.removeItemFromBasket.data.removeItemFromBasket,
+            };
+            this.products = [...this.basket.products];
```

Comment:
Similar to the above comment, I'm not sure what updating the products array here is doing.

Reply:
Will check if we even need to do this.